### PR TITLE
move constructors for graph chunks

### PR DESCRIPTION
**Pull request by karasikov**
Refactoring in graph chunks: use move constructors instead of passing pointers

### CODE REVIEW

File: metagraph/src/graph/representation/succinct/boss_chunk.cpp [hunk 1]
```diff
@@ -138,7 +138,8 @@ BOSS::Chunk::Chunk(uint64_t alph_size, size_t k, bool canonical,
 }
 
 BOSS::Chunk::~Chunk() {
-    utils::remove_temp_dir(dir_);
+    if (!dir_.empty())
+        utils::remove_temp_dir(dir_);
```
Comment (hmusta) [inline metagraph/src/graph/representation/succinct/boss_chunk.cpp:142]: So I'm guessing the logic here is that if the tmp directory is corrupted somehow you want to keep it? Maybe clarify this with a comment.

File: metagraph/src/graph/representation/succinct/boss_merge.cpp [hunk 2]
```diff
@@ -179,27 +179,23 @@ BOSS::Chunk* merge_blocks_to_chunk(const std::vector<const BOSS*> &graphs,
     if (verbose)
         print_bin_stats(bins);
 
-    std::vector<BOSS::Chunk*> blocks(bins.front().size() - 1, NULL);
```
Comment (hmusta) [inline metagraph/src/graph/representation/succinct/boss_merge.cpp:182]: I guess you weren't using this vector before anyway? The only reference to it that I see is getting its size later.

**Review by hmusta — APPROVED** [review]

File: metagraph/src/graph/representation/succinct/boss_chunk.cpp (hunk 1 above)
Comment (karasikov) [reply metagraph/src/graph/representation/succinct/boss_chunk.cpp:142]: Not exactly. Now we can have uninitialized chunks, where dir_ is empty, and thus can't call blindly call `utils::remove_temp_dir` which can only be called for directories created by `utils::create_temp_dir`.

File: metagraph/src/graph/representation/succinct/boss_merge.cpp (hunk 2 above)
Comment (karasikov) [reply metagraph/src/graph/representation/succinct/boss_merge.cpp:182]: Yeah, this was redundant, I noticed this when was changing these lines.

File: metagraph/src/graph/representation/succinct/boss_chunk.cpp (hunk 1 above)
Comment (karasikov) [reply metagraph/src/graph/representation/succinct/boss_chunk.cpp]: I added a comment

**Pull request merged**